Fixed IPC_PRIVATE handling for SysV Shm.

karya0 opened the pull request:
IPC_PRIVATE can [now] be called multiple times and it will in turn create new handlers every time.

Also updated sysv-shm1 test to create shm segments with IPC_PRIVATE key.

Thanks @gkliu for reporting the bug and creating an initial PR (#1191) with the fix.

gc00 commented on include/shareddata.h:
```diff
@@ -198,7 +198,8 @@ void updateDlsymOffset(int32_t dlsymOffset, int32_t dlsymOffset_m32 = 0);
 
-int32_t getRealIPCId(int type, int32_t virt);
+int32_t getRealIPCId(int type, int32_t virt,
+                     bool insertIfNotFound = true);
```
If I'm reading this right, the default value for 1insertIfNotFound` is 'true', and there is one invocation in which we make the argument explicit, inside `SysVShm::virtualToRealKey(key_t k)`.  But that invocation also specifies 'true' for this optional argument.

Is there any case where the optional argument would be false?  If not, then let's just hardwire the code to always insert, if not found. 

I also prefer removing the optional argument, because we eliminiate one of the non-local issues in the design of C++.  If we remove the optional arg, the reader no longer has to look at the .h file to determine if there is a missing default argument, and what is the default value.

karya0 replied:
That's a good catch. It is supposed to be `false` to ensure we don't break any older scenarios with plugins like MANA. I have fixed it now.

gc00 reviewed (changes requested):
@karya0  (and  @gkliu), Thanks for this PR.  It's an important fix.

@karya0 , I have one suggestion:  Since the optional argument is always 'true', why not remove the optional argument?  But if you have a reason for keeping the optional arg, let me know, and I'll still approve it.

gc00 commented:
I'm not sure if I'm the one who should be replying after an update.  Just in case, this is still the only concern that I have:
> @karya0 , I have one suggestion: Since the optional argument in SharedData::getRealIPCId() is always 'true', why not remove the optional argument? But if you have a reason for keeping the optional arg, let me know, and I'll still approve it.

karya0 commented:
> I'm not sure if I'm the one who should be replying after an update.  Just in case, this is still the only concern that I have:
> 
> > @karya0 , I have one suggestion: Since the optional argument in SharedData::getRealIPCId() is always 'true', why not remove the optional argument? But if you have a reason for keeping the optional arg, let me know, and I'll still approve it.

@gc00 : I already fixed a typo and added a response to your comment about the optional argument. Please take a look and approve if it looks good.

gc00 reviewed (approved):
Thanks.  I have no mroe comments.
LGTM.
